homeassistant/components/enocean/config_flow.py

```diff
@@ -0,0 +1,109 @@
+
+    async def async_step_user(self, user_input=None):
+        """Handle an EnOcean config flow start.
+
+        This flow will be used when the user sets up the ENOcean integration.
+        It uses a state machine to walk through the different steps of the
+        configuration.
+        """
+        if self._async_current_entries() or DOMAIN in self.hass.data:
+            return self.async_abort(reason="dongle_already_configured")
+
+        if user_input is None:
+            config_state = DongleSetupStates.SELECT_AUTO
+        elif user_input[CONF_DEVICE] == self.MANUAL_PATH_VALUE:
+            config_state = DongleSetupStates.SELECT_MANUAL
+        else:
+            config_state = DongleSetupStates.VALIDATE
+
+        return await self.set_dongle_setup_state(config_state, user_input)
+
+    async def set_dongle_setup_state(self, new_state, user_input=None, errors=None):
+        """Change the current state of the dongle setup state machine (re-entrant method)."""
+        if new_state == DongleSetupStates.SELECT_MANUAL:
+            LOGGER.debug("Config step: select manual with errors=%s", errors)
+            default_value = None
+            if (
+                user_input is not None
+                and CONF_DEVICE in user_input
+                and user_input[CONF_DEVICE] != self.MANUAL_PATH_VALUE
+            ):
+                default_value = user_input[CONF_DEVICE]
+            return self.async_show_form(
+                step_id="init",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_DEVICE, default=default_value): str}
+                ),
+                errors=errors,
+            )
+
+        if new_state == DongleSetupStates.SELECT_AUTO:
+            LOGGER.debug("Config step: select auto")
+            bridges = EnOceanDongle.detect()
+            if len(bridges) == 0:
+                return await self.set_dongle_setup_state(
+                    DongleSetupStates.SELECT_MANUAL, user_input
+                )
+
+            bridges.append(self.MANUAL_PATH_VALUE)
+            return self.async_show_form(
+                step_id="init",
+                data_schema=vol.Schema({vol.Required(CONF_DEVICE): vol.In(bridges)}),
+            )
+
+        if new_state == DongleSetupStates.VALIDATE:
+            LOGGER.debug("Config step: validate")
+            dongle_path = user_input[CONF_DEVICE]
+            if path.exists(dongle_path) and not path.isdir(dongle_path):
+                return await self.set_dongle_setup_state(
+                    DongleSetupStates.CREATE_ENTRY, user_input
+                )
+
+            errors = {CONF_DEVICE: ERROR_INVALID_DONGLE_PATH}
+            return await self.set_dongle_setup_state(
+                DongleSetupStates.SELECT_MANUAL, user_input, errors
+            )
+
+        if new_state == DongleSetupStates.CREATE_ENTRY:
+            LOGGER.debug("Config step: create entry")
+            await self.async_set_unique_id(user_input[CONF_DEVICE])
```

Comment:
Is the path really unique and stable? If the USB is connected a second time the device path might change, but the device is still the same. I don't think we can use the path and our unique_id api together.

We only allow a single config entry and there's no discovery step so I don't think we need a unique_id for the config entry.

Reply:
Ok, removed the call to set_unique_id